lib/compat.php

```diff
@@ -8,6 +8,45 @@
  * @package gutenberg
  */
 
+if ( ! function_exists( 'register_block_type_from_metadata' ) ) {
```

Comment:
Is it safe to assume that, at the point this function would be implemented in core, the load order would be such that this code would always execute _after_ the core function has been defined, and not the other way around?

(I believe it is, but just raising out of an abundance of caution)

Reply:
Good call. This seems to belong in wp-includes/blocks.php, will be loaded quite early, before mu_plugins.

Reply:
There are multiple occurrences in the plugin where a similar guard clause is added before backporting to WordPress core. I think it's safe to assume so based on the prior art. Examples:
- https://github.com/WordPress/gutenberg/pull/13408
- https://github.com/WordPress/gutenberg/pull/10754

@mcsf can you confirm?

Reply:
My memory fails me on the details, but I seem to recall instances where we had taken some precautions that were enough during normal operation but might not be enough during a WordPress core upgrade (particularly scary before the WSoD protections).

> Good call. This seems to belong in wp-includes/blocks.php, will be loaded quite early, before mu_plugins.

@azaozz, any caveats that you can think of concerning upgrades?

Reply:
Generally, not for plugin updates and when WP is updated from PHP (the built-in updater). There is some risk when svn or git is used. Loading WP while the files are being downloaded and overwritten or copied could trigger fatal errors (some files are updated, others not yet).